# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../sane-frontends/scan-button.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Changelog	18 Jan 2006 09:20:59 -0000	1.198
Changelog	1 Mar 2006 11:05:51 -0000
2006-03-01  Rene Rebe <rene@exactcode.de>

	* src/xscanimage.c: Fixed scan_button sensitivity after a failure
	  at sane_start time, e.g. no documents. Formerly the button was left
	  inactive.

Index: src/xscanimage.c
===================================================================
RCS file: /cvsroot/sane/sane-frontends/src/xscanimage.c,v
retrieving revision 1.38
diff -u -r1.38 xscanimage.c
--- ./src/xscanimage.c	18 Jan 2006 09:20:29 -0000	1.38
+++ ./src/xscanimage.c	1 Mar 2006 11:05:56 -0000
@@ -4,6 +4,7 @@
    Andreas Beck <becka@sunserver1.rz.uni-duesseldorf.de>
    Tristan Tarrant <ttarrant@etnoteam.it>
    David Mosberger-Tang <davidm@azstarnet.com>
+   Rene Rebe <rene@exactcode.de>
 
    Copyright (C) 1997, 1998 Andreas Beck, Tristan Tarrant, and David
    Mosberger
@@ -1092,6 +1093,7 @@
   if (status != SANE_STATUS_GOOD)
     {
       gsg_set_sensitivity (dialog, TRUE);
+      gtk_widget_set_sensitive (scan_win.scan_button, TRUE);
       snprintf (buf, sizeof (buf), "Failed to start scanner: %s",
 		sane_strstatus (status));
       gsg_error (buf);
@@ -1112,6 +1114,7 @@
   if (scan_win.param.lines == -1)
     {
       gsg_set_sensitivity (dialog, TRUE);
+      gtk_widget_set_sensitive (scan_win.scan_button, TRUE);
       snprintf (buf, sizeof (buf), "Hand-Scanner mode not supported");
       gsg_error (buf);
       scan_done ();
@@ -1153,6 +1156,7 @@
 	      if (scan_win.param.depth > 8)
 		{
 		  gsg_set_sensitivity (dialog, TRUE);
+		  gtk_widget_set_sensitive (scan_win.scan_button, TRUE);
 		  snprintf (buf, sizeof (buf),
 			    "Separate channel transfers are not supported "
 			    "with %d bits/channel.", scan_win.param.depth);
@@ -1197,6 +1202,7 @@
       if (scan_win.param.depth > 8)
 	{
 	  gsg_set_sensitivity (dialog, TRUE);
+	  gtk_widget_set_sensitive (scan_win.scan_button, TRUE);
 	  snprintf (buf, sizeof (buf), "The Gimp doesn't support images "
 		    "with %d bits/channel.", scan_win.param.depth);
 	  gsg_error (buf);
